base/tuple.jl

```diff
@@ -105,7 +105,8 @@ julia> ntuple(i -> 2*i, 4)
 ```
 """
 function ntuple(f::F, n::Integer) where F
-    t = n <= 0  ? () :
+    (n >= 0) || throw(DomainError())
```

Comment:
Seconding @pabloferz's recommendation. Using the `Val{N}` methods is not always possible, and `ntuple`'s performance is important. Best!